src/main/java/com/amazonaws/services/kinesis/leases/impl/LeaseCoordinator.java

```diff
@@ -228,6 +228,29 @@ public LeaseCoordinator(ILeaseManager<T> leaseManager,
                 maxLeasesToStealAtOneTime));
     }
 
+    public LeaseCoordinator(LeaseTaker<T> leaseTaker,
```

Comment:
New constructors need to be chained. Not chaining them causes too much overhead while trying to maintain it.

Reply:
Yes, I agree. However, most of the params are used to initialize LeaseTaker and LeaseRenewer which are being passed into this constructor, so kinda redundant.

Reply:
Not sure what you mean. We need both LeaseTaker and LeaseRenewer injectable as-is publicly (from adapter) - can't be created in the constructor.